providers/shared/components/backupstore/id.ftl

```diff
@@ -0,0 +1,217 @@
+                        "Types" : INTEGER_TYPE
+                    },
+
+                ]
+            },
+        ]
+/]
+
+[@addComponentDeployment
+    type=BACKUPSTORE_COMPONENT_TYPE
+    defaultGroup="solution"
+/]
+
+[@addChildComponent
+    type=BACKUPSTORE_RULE_COMPONENT_TYPE
+    parent=BACKUP_COMPONENT_TYPE
+    childAttribute="Rules"
+    linkAttributes="Rule"
+    properties=
+        [
+            {
+                "Type"  : "Description",
+                "Value" : "Defines a schedule of backups and the affected targets"
+            }
+        ]
+    attributes=
+        [
+            {
+                "Names" : "Enabled",
+                "Description" : "Permit the rule to be considered",
+                "Types" : BOOLEAN_TYPE,
+                "Default" : true
+            },
+            {
+                "Names" : "Order",
+                "Description" : "The order in which rules are processed",
+                "Types" : NUMBER_TYPE,
+                "Default" : 100
+            },
+            {
+                "Names" : "Schedule",
+                "Description" : "When to produce snapshots"
+                "Children" : [
+                    {
+                        "Names" : "Expression",
+                        "Description" : "UTC based cron schedule to create backup snapshots"
+                        "Types" : STRING_TYPE
+                    },
+                    {
+                        "Names" : "PointInTimeSupport",
+                        "Description" : "Enable ability for point-in-time restoration"
+                        "Types" : BOOLEAN_TYPE,
+                        "Default" : false
+                    }
+                ]
+            },
+            {
+                "Names" : "StartWindow",
+                "Description" : "Period in minutes in which snapshot must start",
+                "Types" : NUMBER_TYPE
+            },
+            {
+                "Names" : "FinishWindow",
+                "Description" : "Period in minutes after a snapshot has started in which it must finish",
+                "Types" : NUMBER_TYPE
+            },
+            {
+                "Names" : "Lifecycle",
+                "Children" : [
+                    {
+                        "Names" : "Expiration",
+                        "Types" : [NUMBER_TYPE],
+                        "Description" : "Days until snapshot is deleted. Must be greater than Offline (if provided)",
+                        "Default" : "_operations"
+                    },
+                    {
+                        "Names" : "Offline",
+                        "Types" : [NUMBER_TYPE],
+                        "Description" : "Days until backup is moved to longer term storage (if supported)",
+                        "Default" : "_operations"
+                    }
+                ]
+            },
+            {
+                "Names" : "Targets",
```

Comment:
Can I use all three at once? If not then might need a TargetType or something like that where you specify for the rule which one is used.

Reply:
MatchesStore just controls conditions so doesn't actually select any targets. 

By default I add a match all if no tag support or specific resources are provided.